python/triton/compiler.py

```diff
@@ -1294,6 +1294,7 @@ def format_of(ty):
         ptr_info.valid = false;
     }}
     ptr_info.dev_ptr = dev_ptr;
+    Py_DECREF(ret);  // Thanks ChatGPT!
```

Comment:
Note that both returns above would leak it the same way.
It only happens when an error is raised so less common but it will have the same behavior.

Reply:
I think the above return only returns if `ret` is a nullptr right?

Reply:
No, the first one is if the `data_ptr()` function returned something that cannot be interpreted as a Python number.
And the second one is if the conversion from the python number to a c++ unsigned long long failed.